src/parameterCfg.m

```diff
@@ -72,17 +68,39 @@
     para.referencePoint = [3,3,2];
 end
 
-% This is selection of planes/nodes by distance. r = 0 means that there is
+% This is selection of planes/nodes by distance. r = inf means that there is
 % no limitation (Default).
-para = fieldToNum(para, 'selectPlanesByDist', [], 0);
+para = fieldToNum(para, 'selectPlanesByDist', [], inf);
 
 % Switch to turn ON or OFF the Qausi dterministic module
 % 1 = ON, 0 = OFF (Default)
-para = fieldToNum(para, 'switchQDGenerator', [0,1], 0);
+para = fieldToNum(para, 'switchDiffuseComponent', [0,1], 0);
+
+% Switch to consider only diffuse components up to 
+% diffusePathGainThreshold (in dB) below the deterministic ray. This switch 
+% is only used for NIST measurement based scenarios. Default value is set 
+% as -inf which means software does not discard any diffuse components
+para = fieldToNum(para, 'diffusePathGainThreshold', [], -inf);
+
+% Qausi deterministic model
+% nistMeasurements : model based on NIST measurements (Default), 
+% tgayMeasurements : model based on TGay channel document measurements. 
+if ~isfield(para, 'switchQDModel')
+    warning(strcat('Q-D model is not defined in paraCfgCurrent.txt. ',...
+        ' Thus, considering nistMeasurements (default) switchQDModel.'));
+    para.switchQDModel = 'nistMeasurements';
+end
+
 
 % Order of reflection.
 % 1 = multipath until first order, 2 = multipath until second order (Default)
 para = fieldToNum(para, 'totalNumberOfReflections', [], 2);
+if strcmp(para.switchQDModel,'tgayMeasurements') ...
```

Comment:
any reason why are you constraining the reflection order? Why can't the user try higher reflection orders?

Reply:
Yes we are calculating angle of incident till second order of reflection in TGAY model. This has been clearly described in the document

Reply:
ok thank for the clarification.
In this case, i'm not sure that a warning is enough.
I had discussion from last year-ish with @tangnist (not sure why github won't let me mention him directly..) were we had different opinions on whether we should punish a user for a poor decision (e.g., with an error) or correct the poor behavior assuming that you know better than him and possibly notifying him (e.g., with a warning).
There a pros and cons for both sides, if a warning is ok for you guys then i'm fine with it, I just wanted to point this out

Reply:
He did not pointed out anything in documentation so I think he should be fine with the warning.

Reply:
Mattia, I'm still in general more in favor of punishment, but at one point, we need to deliver so I'm fine with it! Moreover, when the behavior is described in the documentation,  at least our user still gets a chance to understand what happened.